s2a.go

```diff
@@ -276,6 +278,52 @@ func (c *s2aTransportCreds) OverrideServerName(serverNameOverride string) error
 	return nil
 }
 
+// TLSClientConfigOptions specifies parameters for creating client TLS config.
+type TLSClientConfigOptions struct{}
+
+// TLSClientConfigFactory defines the interface for a client TLS config factory.
+type TLSClientConfigFactory interface {
+	Build(ctx context.Context, opts *TLSClientConfigOptions) (*tls.Config, error)
+}
+
+// NewTLSClientConfigFactory returns an instance of s2aTLSClientConfigFactory.
+func NewTLSClientConfigFactory(opts *ClientOptions) (TLSClientConfigFactory, error) {
```

Comment:
Can we add tests for this API, similar to what you did in s2av2_e2e_test.go?

Reply:
done, thanks!